### PR TITLE
Port to windows & add CI

### PR DESCRIPTION
**Pull request by lovasoa**
-  makes the code platform-independent, without changing the current behavior on linux.
 - adds a CI job that produces compiled binaries for windows, macos, and linux

### CODE REVIEW

**Comment by Unoqwy** [comment]
Supporting all platforms is definitely a good thing, thank you.

What is the default config path for Windows and MacOS with these changes? On directories-rs example, they show the path for Windows being resolved to `C:\Users\Alice\AppData\Roaming\Foo Corp\Bar App\config`, though in this case it seems like `\Foo Corp` would be omitted (so the full qualified path would be `C:\Users\Alice\AppData\Roaming\Shiny Dir\config\shinydir.toml`), is that right?

**Comment by lovasoa** [comment]
Yes, it will be 
 - `C:\Users\Alice\AppData\Roaming\Shiny Dir\config\shinydir.toml` on windows, and
 - `/Users/Alice/Library/Application Support/Shiny-Dir/shinydir.toml` on MacOS

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -25,21 +35,44 @@ jobs:
...
           ~/.cargo/registry/cache/
           ~/.cargo/git/db/
-          target/            
-        key: ${{ runner.os }}-debug-${{ hashFiles('**/Cargo.lock') }}
-        restore-keys: ${{ runner.os }}-debug-
-    - run: cargo fmt --check
+          target/
+        key: ${{ runner.os }}-check-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: ${{ runner.os }}-check-
+    - run: cargo check --verbose
```
Comment (lovasoa) [inline .github/workflows/rust.yml]: Doesn't cargo test already perform a cargo check ?

Comment (Unoqwy) [reply .github/workflows/rust.yml]: It does, you're right

Comment (Unoqwy) [reply .github/workflows/rust.yml]: I just remove this redundant step on master, thx for pointing it out

**Pull request merged**